sphinxcontrib/images.py

```diff
@@ -157,17 +157,16 @@ def run(self):
         if self.is_remote(self.arguments[0]):
             img['remote'] = True
             if download:
-                img['uri'] = os.path.join('_images', hashlib.sha1(self.arguments[0].encode()).hexdigest())
+                reluri = os.path.join(conf['cache_path'], hashlib.sha1(self.arguments[0].encode()).hexdigest())
+                img['uri'] = os.path.join('/', reluri) # relative to source dir
                 img['remote_uri'] = self.arguments[0]
-                env.remote_images[img['remote_uri']] = img['uri']
-                env.images.add_file('', img['uri'])
+                env.remote_images[img['remote_uri']] = reluri
             else:
                 img['uri'] = self.arguments[0]
                 img['remote_uri'] = self.arguments[0]
         else:
             img['uri'] = self.arguments[0]
             img['remote'] = False
-            env.images.add_file('', img['uri'])
```

Comment:
Is this change needed? I'm not completely up to speed about `env.images` since I've not looked at this code in quite awhile.

Reply:
We don't need to add file by myself -- because we return a image node, it will be processed by [sphinx's env collector](https://github.com/sphinx-doc/sphinx/blob/e0b1e1002b500acc63dfd0806f8095dd6b27037b/sphinx/environment/collectors/asset.py#L93).

Reply:
In contrast to line 163 (`env.images.add_file('', img['uri'])`) seems necessary but line 170 does not, local images are copied to the `_build/html/_images/` without this line. See also my comment here https://github.com/sphinx-contrib/images/pull/23#issuecomment-870083460